### PR TITLE
feat: insights feedback functionality

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/10243

### To test

- [ ] Add [insights feature flag](https://www.notion.so/parabol/How-to-enable-or-remove-a-feature-flag-4b4be190ed674bf8903340d6b5fa50e4), and go to a team where you're the team lead
- [ ] Go to the Insights tab and click "Give us feedback"
- [ ] See that it opens a modal
- [ ] Enter info and click submit. See it stores it in amplitude

### CODE REVIEW

**Comment by nickoferrall** [comment]
Hey @tianrunhe, would you mind reviewing this one? 🙏  Curious to get your thoughts on the `useManualClientSideTrack` hook. I couldn't use the `useClientSideTrack` hook because we don't want to send the data when the modal opens, but it also seemed excessive to create a mutation when we're not updating the db.

Hopefully, `useManualClientSideTrack` could be a helpful pattern.

**Comment by tianrunhe** [comment]
Sorry I didn't quite follow why you need a new hook? Can you explain a little bit more? Thank you!

**Comment by nickoferrall** [comment]
@tianrunhe the `useClientSideTrack` tracks the event when the component loads, so when a modal is opened, for instance, we could track `Modal Opened`. 

But what if we don't want to track the event when the component loads, but when a user does an action, like submitting a form. 

At the moment, I think we can only handle this on the server, but creating a server mutation seems like overkill if you're not updating the db. So I created this hook so that we can track the action once an action happens, like submitting a form.

File: packages/client/hooks/useManualClientSideTrack.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+let eventId = 0
+
+const useManualClientSideTrack = () => {
+  const atmosphere = useAtmosphere()
+
+  const trackEvent = useCallback(
+    (event: string, options: Record<string, any>) => {
+      SendClientSideEvent(atmosphere, event, {
+        ...options,
+        eventId: ++eventId
```
Comment (tianrunhe) [inline packages/client/hooks/useManualClientSideTrack.ts:14]: Why do we need `eventId` here?

File: packages/server/utils/analytics/analytics.ts [hunk 2]
```diff
@@ -709,6 +710,19 @@ class Analytics {
     })
   }
 
+  insightsFeedbackSubmitted = (
+    user: AnalyticsUser,
+    isUseful: boolean | null,
```
Comment (tianrunhe) [inline packages/server/utils/analytics/analytics.ts:715]: -1: it seems `isUseful` would never be `null`?

**Review by tianrunhe — CHANGES_REQUESTED** [review]

File: packages/client/hooks/useManualClientSideTrack.ts (hunk 1 above)
Comment (nickoferrall) [reply packages/client/hooks/useManualClientSideTrack.ts]: We have it in the other hook as there seemed to be a bug where some users were sending it non-stop, and we were trying to debug whether it was happening from that hook or elsewhere. 

If it is happening in the other hook, it's more likely to be related to the `useEffect` which isn't in this hook. I'll remove it, and if the event fires non-stop too, I'll add eventId back.

**Pull request merged**